docs/md/expressions.md

```diff
@@ -0,0 +1,429 @@
+### Editing or Deleting a Custom Column
+
+Once a Custom Column is added, the expression editor can be opened by pressing
+the hamburger button beside the column name. The column may be deleted by
+pressing the X button, but to do so, it must first be removed from the data view
+by unchecking the box beside it.
+
+<img src="/img/expressions/edit-delete-column.png" alt="" />
+
+### Giving a Custom Column a Value
+
+Column Expressions use an "implicit return," in this case meaning the last line
+of the expression will be used as the value. This is called a "return value."
+
+```html
+// My Column Name
+
+2 // Every cell in this column will be `2.00`.
+```
+
+Below you will see our first interactive example.
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Profit", "My Column Name"]'
+  expressions='["// My Column Name\n\n2 // Every cell in this column will be `2.00`."]'
+></perspective-viewer>
+</div>
+
+## Working with Data
+
+### Access an Existing Column's Value
+
+Write an existing column's name in double quotes.
+
+```html
+// My Column Name
+
+"Sales" // Every cell in this column will be identical to "Sales".
+```
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Profit", "My Column Name"]'
+  expressions='["// My Column Name\n\n\"Sales\" // Every cell in this column will be identical to \"Sales\"."]'
+></perspective-viewer>
+</div>
+
+### Variables
+
+While operations may be made directly against a column value (e.g. `"Sales" +
+200`), multiple operations are better organized into variables. Variable
+declarations must end with a semicolon `;`.
+
+```html
+// My Column Name
+
+var incrementedBy200 := "Sales" + 200;
+var half := incrementedBy200 / 2;
+
+half // This returns the value of `half`.
+```
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Profit", "My Column Name"]'
+  expressions='["// My Column Name\n\nvar incrementedBy200 := \"Sales\" + 200;\nvar half := incrementedBy200 / 2;\n\nhalf // This returns the value of `half`."]'
+></perspective-viewer>
+</div>
+
+### Text Values (Strings)
+
+A string is represented with single quotes `var keyphrase := 'Hello World'`.
+
+## Functions
+
+ExprTK and Perspective provide many functions for data manipulation.
+
+> **NOTE**
+>
+> For as full list of capabilities *with autocomplete*, press `Ctrl+Space` in
+> the expression editor.
+
+### Functions Example
+
+Edit the Custom Column and paste the below Column Expression into the editor.
+
+```html
+// My Column Name
+
+var upperCustomer := upper("Customer Name");
+var separator := concat(upperCustomer, ' | ');
+var profitRatio := floor(percent_of("Profit", "Sales")); // Remove trailing decimal.
+var combined := concat(separator, string(profitRatio));
+var percentDisplay := concat(combined, '%');
+
+percentDisplay
+```
+
+<div>
+<perspective-viewer
+  columns='["Customer Name", "Sales", "Profit", "My Column Name"]'
+  expressions='["// My Column Name\n\n0 // REPLACE ALL AND SAVE"]'
+></perspective-viewer>
+</div>
+
+### Strings (Text)
```

Comment:
I appreciate the amount of work this section likely took and there's definitely some good imrpovements to the prose compared to the [auto-generated version I just merged](https://github.com/finos/perspective/blob/master/rust/perspective-viewer/exprtk.md).  However, I'd like the content here to come from the [metadata in the rust src](https://github.com/finos/perspective/blob/master/rust/perspective-viewer/src/rust/exprtk/language.rs#L63), rather than embedded directly in this doc, because:

1) This project is huge and we need to be diligent in auto-generating consistency tasks like this or I will lose my mind.
2) We can assert as a compile-time check that docs exist for new functions that are added or old functions that are changed.
3) I want this information in the intellisense highlights in the actual `monaco` IDE also, and I don't want to have to remember to repeat it in multiple locations.

Any of this content you'd like to move to the metadata (including and especially the examples) would be great